lib/autokey/model.py

```diff
@@ -288,19 +288,25 @@ def get_serializable(self):
             return {"regex": None, "isRecursive": False}
 
     def load_from_serialized(self, data):
-        if isinstance(data, dict): # check needed for data from versions < 0.80.4
-            self.set_window_titles(data["regex"])
-            self.isRecursive = data["isRecursive"]
-        else:
-            self.set_window_titles(data)
+        try:
+            if isinstance(data, dict): # check needed for data from versions < 0.80.4
+                self.set_window_titles(data["regex"])
+                self.isRecursive = data["isRecursive"]
+            else:
+                self.set_window_titles(data)
+        except re.error as e:
```

Comment:
This try block looks like it does nothing?
```python
except re.error as e:
    raise e
```

I think it should silence the error for now like in the dialogue cases. We can add a warning dialog later explaining to the user that it was reset because the saved data was invalid.

Reply:
Yeah, I figure it can't deserialise an erroneous regex if it can't make them in the first place

Reply:
But the user can quit AutoKey and then modify the expression in the settings file and break it.
But AutoKey is generally not hardened against this kind of data manipulation. I just mentioned it because you added the try block to the deserialiser.

Hardening against erroneous save file manipulation can be done in the future, as that user behaviour looks like not being the main source of crashes and issues. I don’t consider it to be a top priority at the moment.

Reply:
I recognised that, but wrote it off since if you're manipulating files like that you're prepared for the consequences. It's 'unsupported' behavior.